### ISSUE TITLE
Upload image into listfield and tie to a model 

### ISSUE THREAD

**Issue by cybertoast** [flask-admin/flask-admin#356]
I'd like to save my assets and the related collection separately:

```
class Assets(db.Document):
  name = db.StringField()
  uri  = db.StringField()

class AssetsView(ModelView):
  form_overrides = {'avatars': SelectMultipleField}
  form_args = {'avatars': { 'base_path': '/tmp/' }}

class Users(db.Document):
  name = db.StringField()
  avatars = db.ListField(db.ReferenceField('Assets'))

```

This allows me to upload an image into the Assets collection and  have it available through the select-list in the Users collection, as a 2-step process.

In order to make the workflow simpler, the upload should really just allow for the file to be uploaded into the User document's avatars list. I can do this by creating a dummy form value, such as "Add avatar", which I can then have as a modal popup, or a separate page. But I wonder if something like this is already in place since this seems like a common workflow.

Basically I'm asking what best practice using Flask-admin for this sort of dynamic addition of related content is (and it could be for any ReferenceField-type content).

Thanks much.
